Add extractDate to Content Extractors.

aborsu opened the pull request:
Returns an Option[DateTime] tries to get the date from the following
tags in that order.
  meta property=article:published_time (openGraph)
  meta name=DCTERMS.created (Dublin)
  time class=dt-published published entry-date
  time itemprop=datePublished
  meta name=DisplayDate
  meta name=date
Finally if all fails it return None

Commit:
Move extractDate to ContentExtractor object

Returns an Option[DateTime]
tries         OpenGraph article:published_time
falls back on Dublin DCTERMS.created
falls back on time class=dt-published published entry-date | itemprop=datePublished
falls back on meta name=DisplayDate | name=date

Unit tests have been modified accordingly.

albertpastrana commented:
Hi @aborsu,

First of all thank you very much for the contribution, it's an excellent idea :-).

Actually, after seeing this we've realised that we should have moved the `extractDate` method from `Gander` to `ContentExtractor` in the first place.

Before accepting the PR though, we believe there could be a couple of changes:
- Consistent indentation (the `orElse` are not nested) and no code commented (the `println`)
- `Gander` is still using its own `extractDate` but should use the new one instead
- It would be great to have some tests that illustrate the use cases. That would help a lot to understand, for example, the reasoning behind this line: `x.split("Z").head.replaceAll("/","-"))`

Let us know if you need any help.

aborsu commented:
@albertpastrana 
Hello, `.split("Z").head` has been removed. 
It's use came from `http://www.dailymail.co.uk/news/article-486484/A-spectacular-destruction-How-email-led-downfall-barrister-all.html` but I saw that the site not only fails to encode the date correctly but also missuses the `publish_date attribute` as it gives the current date.

Regarding `replaceAll("/","-")`. It corrects the input from bbc, I believe the chance of false positive with a replacement like this is small. In the worst case, it will simply return `None`. I've added a comment about it.

The `extractDate` from the Gander object has been replaced by the new one and the unit tests changed accordingly.

I'm gald that you are accepting my contribution.
Is json-ld support somethin you are considering?

albertpastrana commented:
Hi @aborsu,

First of all, sorry for answering so late but it looks like I missed the email from github about this change.

Will merge it now and will generate a new version once we do another change with a small issue we've found.

Regarding json-ld support, what's the use case you are thinking about?
